crates/hir_ty/src/infer/expr.rs

```diff
@@ -1041,11 +1046,13 @@ impl<'a> InferenceContext<'a> {
         for (_id, param) in def_generics.iter_parent() {
             match param {
                 TypeOrConstParamData::TypeParamData(_) => {
-                    substs.push(self.table.new_type_var());
+                    substs.push(GenericArgData::Ty(self.table.new_type_var()).intern(Interner));
                 }
-                TypeOrConstParamData::ConstParamData(_) => {
+                TypeOrConstParamData::ConstParamData(data) => {
                     // FIXME: here we should do something else
-                    substs.push(self.table.new_type_var());
+                    let ty = self.make_ty(&data.ty);
```

Comment:
`make_ty` is wrong here, since it resolves the type with the resolver of the function we're inferring. This should use the `const_param_ty` query.